distributed/protocol/__init__.py

```diff
@@ -117,3 +117,34 @@ def _register_cudf():
 def _register_cuml():
     with ignoring(ImportError):
         from cuml.comm import serialize
+
+
+import collections.abc
+
+import dask
+
+
+class TypeCompressor(collections.abc.MutableMapping):
+    def __init__(self, compressions):
+        self.storage = {}
+        self.compressions = compressions
+
+    def __setitem__(self, key, value):
+        header, frames = serialize(value)
+        self.storage[key] = {"header": header, "data": maybe_compress(frames)}
```

Comment:
Can this just store a tuple of `(header, data)`? Seems more natural than a dict with always 2 keys.